src/Compilers/Core/Portable/Operations/ControlFlowGraphBuilder.cs

```diff
@@ -3104,11 +3108,215 @@ public override IOperation VisitObjectCreation(IObjectCreationOperation operatio
+                        return Visit(assignmentOperation);
+
+                    case OperationKind.ArrayElementReference:
+                        // PROTOTYPE(dataflow): Add support
+
+                    default:
+                        throw ExceptionUtilities.UnexpectedValue(left.Kind);
+                }
+
+                return new SimpleAssignmentExpression(left, assignmentOperation.IsRef, right, semanticModel: null, assignmentOperation.Syntax,
+                                                      assignmentOperation.Type, assignmentOperation.ConstantValue, assignmentOperation.IsImplicit);
+            }
+
+            void handleMemberInitializer(IMemberInitializerOperation memberInitializer)
+            {
+                // We explicitly do not push the initialized member onto the stack here. We visit the initialized member to get the implicit receiver that will be substituted in when an
+                // IInstanceReferenceOperation with InstanceReferenceKind.ImplicitReceiver is encountered. If that receiver needs to be pushed onto the stack, its parent will handle it.
+                // In member initializers, the code generated will evaluate InitializedMember multiple times. For example, if you have the following:
+                //
+                // class C1
+                // {
+                //   public C2 C2 { get; set; } = new C2();
+                //   public void M()
+                //   {
+                //     var x = new C1 { C2 = { P1 = 1, P2 = 2 } };
+                //   }
+                // }
+                // class C2
+                // {
+                //   public int P1 { get; set; }
+                //   public int P2 { get; set; }
+                // }
+                //
+                // We generate the following code for C1.M(). Note the multiple calls to C1::get_C2().
+                //   IL_0000: nop
+                //   IL_0001: newobj instance void C1::.ctor()
+                //   IL_0006: dup
+                //   IL_0007: callvirt instance class C2 C1::get_C2()
+                //   IL_000c: ldc.i4.1
+                //   IL_000d: callvirt instance void C2::set_P1(int32)
+                //   IL_0012: nop
+                //   IL_0013: dup
+                //   IL_0014: callvirt instance class C2 C1::get_C2()
+                //   IL_0019: ldc.i4.2
+                //   IL_001a: callvirt instance void C2::set_P2(int32)
+                //   IL_001f: nop
+                //   IL_0020: stloc.0
+                //   IL_0021: ret
+                //
+                // We therefore visit the InitializedMember to get the implicit receiver for the contained initializer, and that implicit receiver will be cloned everywhere it encounters
+                // an IInstanceReferenceOperation with ReferenceKind InstanceReferenceKind.ImplicitReceiver
+
+                IOperation instance = Visit(memberInitializer.InitializedMember);
```

Comment:
>IOperation instance = Visit(memberInitializer.InitializedMember); [](start = 16, length = 65)

Do we have a test covering a control flow inside ```memberInitializer.InitializedMember``` at this point? Inside an index and elsewhere perhaps in error cases. It would be especially interesting to test with coalesce expression.